src/exception_utils.rs

```diff
@@ -245,31 +245,15 @@ pub fn exception_data_abort_access_is_sign_ext() -> bool {
 macro_rules! save_regs_to_stack {
     () => {
         "
-        sub     sp, sp, 34 * 8
-        stp     x0, x1, [sp]
-        stp     x2, x3, [sp, 2 * 8]
-        stp     x4, x5, [sp, 4 * 8]
-        stp     x6, x7, [sp, 6 * 8]
-        stp     x8, x9, [sp, 8 * 8]
-        stp     x10, x11, [sp, 10 * 8]
-        stp     x12, x13, [sp, 12 * 8]
-        stp     x14, x15, [sp, 14 * 8]
-        stp     x16, x17, [sp, 16 * 8]
-        stp     x18, x19, [sp, 18 * 8]
-        stp     x20, x21, [sp, 20 * 8]
-        stp     x22, x23, [sp, 22 * 8]
-        stp     x24, x25, [sp, 24 * 8]
-        stp     x26, x27, [sp, 26 * 8]
-        stp     x28, x29, [sp, 28 * 8]
-
-        mov     x1, sp
-        add     x1, x1, #(0x110)
-        stp     x30, x1, [sp, 30 * 8]
-        mrs     x10, elr_el2
-        mrs     x11, spsr_el2
-        stp     x10, x11, [sp, 32 * 8]
-
-        add    sp, sp, 34 * 8"
+        sub     sp, sp, 12 * 8
+        // save old context (callee-saved registers)
+        stp     x29, x30, [sp, 10 * 8]
+        stp     x27, x28, [sp, 8 * 8]
+        stp     x25, x26, [sp, 6 * 8]
+        stp     x23, x24, [sp, 4 * 8]
+        stp     x21, x22, [sp, 2 * 8]
+        stp     x19, x20, [sp]
+        add    sp, sp, 12 * 8"
```

Comment:
Only, save x30 ?

Reply:
I try to save x30 only, but failed.

Reply:
> wo should save el2 ctx, then when we restore el2 ctx latter, we can excute "ret", jump to el2 old state.

Yes, but there is no need to do this ugly `sub sp` and `add sp` operation,  
I think we can just simply `sub sp` and keep sp as it is

Reply:
> > wo should save el2 ctx, then when we restore el2 ctx latter, we can excute "ret", jump to el2 old state.
> 
> Yes, but there is no need to do this ugly `sub sp` and `add sp` operation, I think we can just simply `sub sp` and keep sp as it is

We must change sp, because callee-saved regs must save in stack.